provider/app.py

```diff
@@ -141,6 +149,24 @@ def restoreTriggers():
         createAndRunConsumer(triggerFQN, triggerDoc, record=False)
 
 
+def getMissingPostFields(fields):
+    missing = []
+    required = ['brokers', 'topic', 'triggerURL', 'isMessageHub']
+
+    if fields is None:
+        missing = required
+    else:
+        missing.extend([i for i in required if i not in  fields])
+
+        if 'isMessageHub' in fields and fields['isMessageHub'] == True:
+            missing.extend([i for i in ['username', 'password'] if i not in fields])
```

Comment:
Instead of checking the params twice, how about optionally adding `username` and `password` to the `required` array, and then only checking once?